pkg/types/aws/machinepool.go

```diff
@@ -35,6 +35,12 @@ type MachinePool struct {
 	// +optional
 	IAMRole string `json:"iamRole,omitempty"`
 
+	// IAMProfile is the name of the IAM instance profile to use for the machine.
+	// Leave unset to have the installer create the IAM Profile on your behalf.
```

Comment:
Should the opposite warning be applied to the IamRole?

Reply:
No. When users set`IAMRole` the Installer creates a profile.

Reply:
Unless I didn't understand your suggestion here. A profile should include a role (we validate this during install-config validation).

Reply:
We clarified that you meant `"// Cannot be specified together with iamRole."`. I'll add it in the next rebase.

Reply:
Done